syntax = "proto3";

option java_multiple_files = true;
option java_package = "lune.grpc.extractor";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package extractor;

service Extractor {
  rpc extract (ExtractorRequest) returns (ExtractorReply) {}
}

message ExtractorRequest {
  UploadInfo upload_info = 1;
  string process_rule_mode = 2;
  string datasource_type = 3;
  string document_model = 4;
}

message UploadInfo {
  string key = 1;
  string tenant_id = 2;
  string created_by = 3;
}

message ExtractorReply {
  sint32 code = 1;
  string msg = 2;
  repeated DocumentReply documents = 3;
}

message DocumentReply {
  string page_content = 1;
  string provider = 2;
  map<string, string> meta_data =3; 
}